Add custom layout functionality and related methods to Niivue

hanayik opened the pull request:
- Introduced `setCustomLayout`, `clearCustomLayout`, and `getCustomLayout` methods.
- Implemented custom layout handling in `drawSceneCore`. Custom layouts have higher priority than built-in layouts.
- Updated hot-reloadable index.html to include a toggle button for custom layouts during development.

### Example:

```javascript
// Layout format: {sliceType: SLICE_TYPE, position: [left, top, width, height]}
// Units are relative (0-1) to canvas size
const customLayout = [
  // Left 50% - Sag
  {sliceType: SLICE_TYPE.SAGITTAL, position: [0, 0, 0.5, 1.0]},
  // Top right - Cor
  {sliceType: SLICE_TYPE.CORONAL, position: [0.5, 0, 0.5, 0.5]},
  // Bottom right - Ax
  {sliceType: SLICE_TYPE.AXIAL, position: [0.5, 0.5, 0.5, 0.5]}
];
      
  // Apply the custom layout
  nv1.setCustomLayout(customLayout);

  // clear the custom layout and revert to built-in layouts
  nv1.clearCustomLayout()

/*
   * +----------------+----------------+
   * |                |                |
   * |                |     coronal    |
   * |                |                |
   * |                |                |
   * |   sagittal     +----------------+
   * |                |                |
   * |                |     axial      |
   * |                |                |
   * |                |                |
   * +----------------+----------------+
 */
```

<img width="1505" alt="image" src="https://github.com/user-attachments/assets/835544a0-238c-4d9a-a12a-2fa2d38b5d2b" />


List of fixed issues (if they exist):

- closes #1237

neurolabusc commented:
This is very nice. However, you should deduct the size of the colorbar (if present) from the size of the canvas. You can do this with reserveColorbarPanel() to determine canvas pixels required for colorbar.

You can see the issue by setting the `isColorbar` to true: 

```javascript
      var nv1 = new Niivue(defaults)
      await nv1.attachToCanvas(gl1)
      nv1.opts.isColorbar = true
```

hanayik commented:
@neurolabusc , good catch! I have added the space reservation for the colorbar, and also added some checks to ensure the custom layout is a valid layout (e.g. no overlapping tiles, and prevent a tile from extending past the canvas bounds) by accident.

neurolabusc reviewed (approved):
Great feature
